src/main/resources/templates/db_scripts/migrateItemCopyNumberToSingleValue.sql

```diff
@@ -1,12 +1,40 @@
 START TRANSACTION;
--- Make the array of copy numbers as a single value, take the first value from the array
-UPDATE ${myuniversity}_${mymodule}.item AS it
-  SET jsonb = jsonb_set(it.jsonb, '{copyNumber}', it.jsonb#>'{copyNumbers, 0}')
-WHERE it.jsonb->>'copyNumbers' IS NOT NULL AND jsonb_array_length(it.jsonb->'copyNumbers') > 0;
-
--- Remove the copyNumbers property even if it has zero length.
-UPDATE ${myuniversity}_${mymodule}.item AS it
-  SET jsonb = it.jsonb - 'copyNumbers'
-WHERE it.jsonb->'copyNumbers' IS NOT NULL;
+
+UPDATE pg_trigger
+  SET tgenabled = 'D'
+WHERE tgrelid IN (SELECT oid FROM pg_class WHERE relname = 'item')
+  AND tgisinternal IS FALSE
+  AND tgenabled = 'O';
+
+-- Temporary Trigger: perform conditional logic to perform upgrades in one update without a WHERE.
+CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.temp_upgrade_items()
+  RETURNS TRIGGER AS $$
+  BEGIN
+    IF (TG_OP = 'UPDATE' AND NEW.jsonb->'copyNumbers' IS NOT NULL) THEN
+      IF (NEW.jsonb->>'copyNumbers' IS NOT NULL
+        AND jsonb_array_length(NEW.jsonb->'copyNumbers') > 0) THEN
+        NEW.jsonb := jsonb_set(NEW.jsonb, '{copyNumber}', NEW.jsonb#>'{copyNumbers, 0}');
+      END IF;
+      NEW.jsonb := NEW.jsonb - 'copyNumbers';
+    END IF;
+    RETURN NEW;
+  END;
+  $$ LANGUAGE 'plpgsql';
+CREATE TRIGGER temp_upgrade_items_trigger
+  BEFORE UPDATE ON ${myuniversity}_${mymodule}.item
+  FOR EACH ROW EXECUTE PROCEDURE ${myuniversity}_${mymodule}.temp_upgrade_items();
+
+UPDATE ${myuniversity}_${mymodule}.item
+  SET id = id
+WHERE jsonb->'copyNumbers' IS NOT NULL;
+
+DROP TRIGGER temp_upgrade_items_trigger ON ${myuniversity}_${mymodule}.item;
+DROP FUNCTION ${myuniversity}_${mymodule}.temp_upgrade_items;
```

Comment:
Why advantages has the plpgsql trigger function over a SQL CASE?
```
UPDATE ${myuniversity}_${mymodule}.item
SET jsonb = CASE WHEN jsonb_array_length(jsonb->'copyNumbers') > 0
                 THEN jsonb_set(jsonb - 'copyNumbers', '{copyNumber}', jsonb#>'{copyNumbers, 0}')
                 ELSE jsonb - 'copyNumbers'
            END
WHERE jsonb->'copyNumbers' IS NOT NULL;
```

Reply:
Thanks! I wasn't aware CASE was available outside of pgsql.
This greatly simplifies the work and allows me to avoid the undesirable function and trigger.